pxf/pxf-hbase/src/main/java/org/apache/hawq/pxf/plugins/hbase/HBaseFilterBuilder.java

```diff
@@ -71,14 +74,19 @@ public HBaseFilterBuilder(HBaseTupleDescription tupleDescription) {
      * @throws Exception if parsing failed
      */
     public Filter getFilterObject(String filterString) throws Exception {
-        FilterParser parser = new FilterParser(this);
-        Object result = parser.parse(filterString);
+        // First check for NOT, HBase does not support this
+        if (filterString.contains(NOT_OP)) {
+            return null;
+        } else {
+            FilterParser parser = new FilterParser(this);
+            Object result = parser.parse(filterString);
+
+            if (!(result instanceof Filter)) {
+                throw new Exception("String " + filterString + " resolved to no filter");
```

Comment:
Rewrite error message to something like "String couldn't be resolved to any supported filter"?